server/src/main/java/org/opensearch/cluster/metadata/Metadata.java

```diff
@@ -1286,6 +1286,11 @@ public Builder templates(Map<String, IndexTemplateMetadata> templates) {
             return this;
         }
 
+        public Builder removeAllTemplates() {
+            this.templates.clear();
```

Comment:
The way diff publication for remote state works is, if there is any change in the templates, the entire templates object is uploaded. So when applying the diff in remote publication, we should clear the previous templates and then add all templates from the current upload.
For the custom metadata, we maintain a hashmap of Custom type to Custom object. So whenver custom metadata changes, it replaces the current custom object in the hashmap.

Reply:
> we should clear the previous templates and then add all templates from the current upload

Can't we then combine these two operations in a single builder method instead of having a remove and a add?

Reply:
Having separate methods makes it makes it more flexible. 
For example, we may want to retain the existing templates.

Reply:
If we don't have a specific need to just remove all templates then we should remove the method for now and add it later when the need arises.

Reply:
I am adhering to the existing convention of builder methods where single operations are performed by the methods.
See `put` for `IndexMetadata` and `removeAllIndices`

Reply:
```
public Builder putMapping(MappingMetadata mappingMd) {
            mappings.clear();
            if (mappingMd != null) {
                mappings.put(mappingMd.type(), mappingMd);
            }
            return this;
        }
```

Reply:
Ok. Removed this buidler method.